### ISSUE TITLE
swarm mode : docker service create --mount type=bind requires a pre exist src path

### ISSUE THREAD

**Issue by tarinidash** [docker/docker#29219]
Currently --mount type=bind requires a source path to pre exist. If its does not exist it throws an error _invalid bind mount source, source path not found_.

is it by design? Shouldn't it create the path if it does not exist? 

**docker version**

Client:
Version: 1.12.3
API version: 1.24
Go version: go1.6.3
Git commit: 6b644ec
Built:
OS/Arch: linux/amd64

Server:
Version: 1.12.3
API version: 1.24
Go version: go1.6.3
Git commit: 6b644ec
Built:
OS/Arch: linux/amd64

**Comment by justincormack** [comment]
I think this is by design, yes. @stevvooe can you confirm?

**Comment by stevvooe** [comment]
This is by design.

The issue is that creating the file requires the runtime to infer too much about what the new resource should actually be. This can include regular metadata, like owner and group, but also whether or not it is a file.

@cpuguy83 may be able to provide more detail.

**Comment by tarinidash** [comment]
I looks like  _docker run_  supports creating a bind mount if its not available . (https://docs.docker.com/engine/reference/run/#/volume-shared-filesystems). It would be nice if swarm mode can support similar functionality.

**Comment by cpuguy83** [comment]
@tarinidash It does, and purely for historical reasons. It's caused more problems than it's worth. Supporting this on a distributed system will be even so much more worse.

What's the point of the bind if you are just having it create the path?

**Comment by thaJeztah** [comment]
This was by design indeed; we deprecated this feature on `docker run` for docker 1.9, but had to reconsider that deprecation, because removing it would be too much of a breaking change (see https://github.com/docker/docker/pull/21666


Automatically creating the host-path results in a lot of issues that we currently have to deal with for `docker run`. For example, the daemon and client can run on a different host, in which case specifying a local file path that doesn't exist on the host that the daemon runs on, results in an empty path being created on that host.

In a regular swarm mode setup, _all_ nodes are remote, so this would lead to an empty directory being created on the node that a task ends on.

For that reason, we decided to _not_ support this for services.

I'll close this issue, but feel free to continue the discussion
